### ISSUE TITLE
Question about weather

### ISSUE THREAD

**Issue by Ry4nGup14** [XengShi/materialYouNewTab#321]
Possible to allow to use location for weather update rather than sticking to one set location?

**Comment by itz-rj-here** [comment]
> Possible to allow to use location for weather update rather than sticking to one set location?

Your request is not clear or not understandable. If you could properly say what you want to know we would like to help you.

**Comment by Ry4nGup14** [comment]
So sorry about that. What I was trying to say is that under the weather where we have to input a location, is there any way to use the current Location based on GPS to update the weather. For eg, I'm in a part of singapore called punggol, and it will show weather of that part, and in the next hour, tje laptop uses Location services and automatically detects that I have moved to another area example downtown area. These changes happen automatically rather than me inputting a set Location.

**Comment by itz-rj-here** [comment]
> So sorry about that. What I was trying to say is that under the weather where we have to input a location, is there any way to use the current Location based on GPS to update the weather. For eg, I'm in a part of singapore called punggol, and it will show weather of that part, and in the next hour, tje laptop uses Location services and automatically detects that I have moved to another area example downtown area. These changes happen automatically rather than me inputting a set Location.

@prem-k-r can you handle this issue please.

**Comment by prem-k-r** [comment]
> > So sorry about that. What I was trying to say is that under the weather where we have to input a location, is there any way to use the current Location based on GPS to update the weather. For eg, I'm in a part of singapore called punggol, and it will show weather of that part, and in the next hour, tje laptop uses Location services and automatically detects that I have moved to another area example downtown area. These changes happen automatically rather than me inputting a set Location.
> 
> @prem-k-r can you handle this issue please.

![image](https://github.com/user-attachments/assets/5d59560c-ad28-4f84-862d-1f44dec2f192)
how about something like this?

**Comment by itz-rj-here** [comment]
> > > So sorry about that. What I was trying to say is that under the weather where we have to input a location, is there any way to use the current Location based on GPS to update the weather. For eg, I'm in a part of singapore called punggol, and it will show weather of that part, and in the next hour, tje laptop uses Location services and automatically detects that I have moved to another area example downtown area. These changes happen automatically rather than me inputting a set Location.
> > 
> > 
> > @prem-k-r can you handle this issue please.
> 
> ![image](https://private-user-images.githubusercontent.com/60751338/396007061-5d59560c-ad28-4f84-862d-1f44dec2f192.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzMzYyNDYsIm5iZiI6MTczNDMzNTk0NiwicGF0aCI6Ii82MDc1MTMzOC8zOTYwMDcwNjEtNWQ1OTU2MGMtYWQyOC00Zjg0LTg2MmQtMWY0NGRlYzJmMTkyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE2VDA3NTkwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1MDc5OWE1MTAyMDRmOGQ2MDc0ZjFjNDliNjk4MjAzMWFjZmFlZGYzODQyOTAyMGNiZTUxNTI4ZTA2NmNkZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.31LVSuadPAMu9FqqYWKBuX2mZiRTRxO7oB_Pj4gmByQ) how about something like this?

yup just the thing he is requesting

**Comment by prem-k-r** [comment]
Ok I managed to implement the funtionalities.. but is this really needed.. I mean PC uses Wi-Fi or IP-based methods for geolocation.. they are inaccurate..
Had to do lots of changes.. and prompt for location permission when clicked first time on the button

the cons I noticed at the moment is that there is a slight increase in delay to show location..

**Comment by Ry4nGup14** [comment]
@prem-k-r yes, just the thing I was asking for. It's pretty useful for laptop users. Thank you so much.
